File: app/src/main/java/com/example/messengerandroid/fragment/OnboardFragment.java
package com.example.messengerandroid.fragment;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.messengerandroid.MainActivity;
import com.example.messengerandroid.R;
import com.example.messengerandroid.databinding.ActivityOnboardBinding;
import com.example.messengerandroid.databinding.ActivitySignInBinding;


public class OnboardFragment extends Fragment {

    private static final String TAG = "OnboardFragment";

    private ActivityOnboardBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = ActivityOnboardBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Log.d(TAG, "onCreate");
        EdgeToEdge.enable(requireActivity());
        binding.btnNext.setOnClickListener(v -> ((MainActivity) requireActivity()).navigateToSignIn());
    }
}

File: app/src/main/java/com/example/messengerandroid/fragment/OnboardFragment.kt
package com.example.messengerandroid.fragment

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.messengerandroid.MainActivity
import com.example.messengerandroid.databinding.ActivityOnboardBinding

class OnboardFragment : Fragment() {

    companion object {
        private const val TAG = "OnboardFragment"
    }

    private var binding: ActivityOnboardBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = ActivityOnboardBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.d(TAG, "onCreate")

        binding?.btnNext?.setOnClickListener {
            (requireActivity() as MainActivity).navigateToSignIn()
        }

        binding?.btnNext?.setOnClickListener {
            (requireActivity() as MainActivity).navigateToSignIn()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

}
